src/object/srv_obj.c

```diff
@@ -1507,13 +1507,22 @@ obj_local_rw_internal(crt_rpc_t *rpc, struct obj_io_context *ioc,
 			goto post;
 		}
 
-		if (ioc->ioc_coc->sc_props.dcp_csum_enabled) {
+		if (ioc->ioc_coc->sc_props.dcp_csum_enabled &&
+		    daos_csummer_initialized(ioc->ioc_coc->sc_csummer)) {
+			struct daos_csummer csummer = {0};
+
+			rc = daos_csummer_copy_inplace(&csummer, ioc->ioc_coc->sc_csummer);
+			if (rc != 0) {
+				D_ERROR("csummer_copy_inplace failed: "DF_RC"\n", DP_RC(rc));
+				goto post;
+			}
 			rc = csum_add2iods(ioh,
 					   orw->orw_iod_array.oia_iods,
 					   orw->orw_iod_array.oia_iod_nr,
-					   ioc->ioc_coc->sc_csummer,
+					   &csummer,
```

Comment:
What's the difference of using the 'sc_csummer' and on stack 'csummer' here? Maybe you'd explain bit more about the race you mentioned?